### PR TITLE
OCM-12364 | feat: create/cluster validation for hcpsharedvpc flags

### PR DESCRIPTION
**Pull request by hunterkepley**

### CODE REVIEW

File: cmd/create/cluster/cmd.go [hunk 1]
```diff
@@ -2332,7 +2332,39 @@ func run(cmd *cobra.Command, _ []string) {
...
+				r.Reporter.Errorf(hcpSharedVpcFlagOnlyErrorMsg,
+					vpcEndpointRoleArnFlag)
+				os.Exit(1)
+			} else if hcpInternalCommunicationHostedZoneId != "" {
+				r.Reporter.Errorf(hcpSharedVpcFlagOnlyErrorMsg,
+					vpcEndpointRoleArnFlag)
+				os.Exit(1)
+			}
+		}
+		if err != nil {
```
Comment (marcolan018) [inline cmd/create/cluster/cmd.go:2361]: should this check move to "if isHostedCP" block?

File: cmd/create/cluster/cmd.go [hunk 2]
```diff
@@ -2332,7 +2332,39 @@ func run(cmd *cobra.Command, _ []string) {
...
+	isHcpSharedVpc := false
+	route53RoleArn := sharedVPCRoleARN // TODO: Change when fully deprecating old flag name
+	vpcEndpointRoleArn := strings.Trim(args.vpcEndpointRoleArn, " \t")
+	hcpInternalCommunicationHostedZoneId := strings.Trim(args.hcpInternalCommunicationHostedZoneId, " \t")
+	ingressPrivateHostedZoneId := privateHostedZoneID // TODO: Change when fully deprecating old flag name
+
+	anyHcpSharedVpcFlagsUsed := route53RoleArn != "" || vpcEndpointRoleArn != "" ||
+		hcpInternalCommunicationHostedZoneId != "" || ingressPrivateHostedZoneId != ""
+
+	if anyHcpSharedVpcFlagsUsed {
```
Comment (marcolan018) [inline cmd/create/cluster/cmd.go:2345]: imho, it's better to include all these logic into the func validateHcpSharedVpcArgs()

File: cmd/create/cluster/cmd.go [hunk 3]
```diff
@@ -2332,7 +2332,39 @@ func run(cmd *cobra.Command, _ []string) {
...
+			err = validateHcpSharedVpcArgs(route53RoleArn, vpcEndpointRoleArn, ingressPrivateHostedZoneId,
+				hcpInternalCommunicationHostedZoneId)
+		} else {
+			if vpcEndpointRoleArn != "" {
+				r.Reporter.Errorf(hcpSharedVpcFlagOnlyErrorMsg,
+					vpcEndpointRoleArnFlag)
+				os.Exit(1)
+			} else if hcpInternalCommunicationHostedZoneId != "" {
+				r.Reporter.Errorf(hcpSharedVpcFlagOnlyErrorMsg,
+					vpcEndpointRoleArnFlag)
```
Comment (gdbranco) [inline cmd/create/cluster/cmd.go:2357]: wrong flag var

**Pull request merged**